services/auth/internal/app/infrastructure/database/redis.go

```
@@ -0,0 +1,38 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"time"
+
+	"github.com/VovkaGoodwin/microservices-cource/pkg/config"
+	"github.com/redis/go-redis/v9"
+)
+
+func NewRedisDb(cfg *config.Config, log *slog.Logger) (*redis.Client, error) {
+	var rdb *redis.Client
+	var err error
+	options := &redis.Options{
+		Addr:     fmt.Sprintf("%s:%s", cfg.DB.Redis.Host, cfg.DB.Redis.Port),
+		Password: cfg.DB.Redis.Password,
+		DB:       cfg.DB.Redis.Db,
+	}
+
+	for i := 0; i < 100; i++ {
```

Говорил в лекции про fail fast стратегию. Если мы неправильно настроим конфиг редиса, то узнаем что что-то не так только через 500 секунд после деплоя...